server/src/main/java/org/elasticsearch/index/engine/Engine.java

```diff
@@ -177,32 +170,14 @@ public MergeStats getMergeStats() {
     /** Returns how many bytes we are currently moving from heap to disk */
     public abstract long getWritingBytes();
 
+
+    final CompletionStatsCache completionStatsCache = new CompletionStatsCache(() -> acquireSearcher("completion_stats"));
```

Comment:
NB this moves the stats back to the `EXTERNAL` searcher scope; previously they were at `INTERNAL` but I think `EXTERNAL` is more appropriate.

Reply:
I don't think we ever use the segments that are only referenced by the `INTERNAL` searcher for completion suggestions, but AIUI the previous implementation would load up their suggesters anyway. By moving this back to `EXTERNAL` we avoid even loading these unnecessary suggesters until they're externally exposed, so there's no need to account for their memory usage.

I've not confirmed 100% that we never use internal-only segments for completion suggestions, so maybe that assumption is flawed.

Reply:
++, thanks for elaborating, this makes perfect sense.